### PR TITLE
Refactoring and integrating apis

### PR DESCRIPTION
**Pull request by Maianki**
### Files to be checked
- src/components/Card/CategoryCard/CategoryCard.jsx
- src/components/Card/ProductCard/ProductCard.jsx
- src/constants/apiEndPoints.js
- src/hooks/useAxios.js
- src/hooks/useDocumentTitle.js
- src/pages/Cart/Cart.jsx
- src/pages/LandingPage/LandingPage.jsx
- src/pages/ProductPage/ProductPage.js
- src/pages/SignupPage/SignUpPage.jsx

### Files that can be ignored
- src/pages/LandingPage/landingPage.css
- src/index.css
- src/backend/db/products.js
- src/assets/images/index.js
- src/assets/icons/index.js
- src/assets/icons/icon.js

#### Live Preview 
[live preview Link](https://shuttle-shopy-react-git-refactoring-and-integrat-6dab1b-maianki.vercel.app/)

### CODE REVIEW

**Review by srishtigautam21 — APPROVED** [review]
The code is clean and well understandable
The UI looks great
Good to merge

File: src/pages/SignupPage/SignUpPage.jsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
 import { Link } from "react-router-dom";
+import { useDocumentTitle } from "../../hooks";
 import "./sign-up-page.css";
 import { BiEye, BiEyeSlash } from "../../assets/icons";
 
 export function SignUpPage() {
+  useDocumentTitle("Signup");
   return (
     <div classNameName='signup-container'>
       <Navbar />
```
Comment (abhisheksahani-nova) [inline src/pages/SignupPage/SignUpPage.jsx:12]: Generally signup page should not have Navbar

**Review by abhisheksahani-nova — APPROVED** [review]
Code looks great , good work

File: src/pages/SignupPage/SignUpPage.jsx (hunk 1 above)
Comment (Maianki) [reply src/pages/SignupPage/SignUpPage.jsx:12]: Thanks for reviewing. I have kept it till the time I add authentication.

**Pull request merged**